subsys/nfc/lib/platform.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <clock_control.h>
+#include <device.h>
+#include <nrfx_nfct.h>
+#include <nrfx_timer.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(nfc_platform, CONFIG_NFC_PLATFORM_LOG_LEVEL);
+
+struct device *clock;
+
+static void clock_handler(struct device *dev, void *user_data)
+{
+	/* Activate NFCT only when HFXO is running */
+	nrfx_nfct_state_force(NRFX_NFCT_STATE_ACTIVATED);
+}
+
+
+static struct clock_control_async_data clock_ctrl = {
+	.cb = clock_handler
+};
+
+
+nrfx_err_t nfc_platform_setup(void)
+{
+	clock = device_get_binding(DT_INST_0_NORDIC_NRF_CLOCK_LABEL "_16M");
+	__ASSERT_NO_MSG(clock);
+
+	IRQ_DIRECT_CONNECT(NFCT_IRQn, CONFIG_NFCT_IRQ_PRIORITY,
+			   nrfx_nfct_irq_handler, 0);
+	IRQ_DIRECT_CONNECT(TIMER4_IRQn, CONFIG_NFCT_IRQ_PRIORITY,
+			   nrfx_timer_4_irq_handler, 0);
```

Comment:
The timer instance number should be made configurable. Though, perhaps not necessarily in this PR.

Reply:
I agree with you. I would like to do it in the other PR.